File: backend/src/HowAreMyFinances.Api/Models/Expense.cs
namespace HowAreMyFinances.Api.Models;

public sealed record Expense(
    Guid Id,
    Guid UserId,
    Guid MonthId,
    Guid CategoryId,
    string ItemName,
    decimal Amount,
    string? Vendor,
    DateOnly ExpenseDate,
    string? Comment,
    bool IsRecurringInstance,
    bool IsCompleted,
    DateTime CreatedAt,
    DateTime UpdatedAt
);

public sealed record ExpenseWithCategory(
    Guid Id,
    Guid UserId,
    Guid MonthId,
    Guid CategoryId,
    string ItemName,
    decimal Amount,
    string? Vendor,
    DateOnly ExpenseDate,
    string? Comment,
    bool IsRecurringInstance,
    bool IsCompleted,
    string CategoryName,
    string? CategoryIcon,
    DateTime CreatedAt,
    DateTime UpdatedAt
);

public sealed record CreateExpenseRequest(
    string ItemName,
    decimal Amount,
    Guid CategoryId,
    string? Vendor,
    DateOnly ExpenseDate,
    string? Comment,
    bool IsCompleted = true
);

public sealed record UpdateExpenseRequest(
    string? ItemName,
    decimal? Amount,
    Guid? CategoryId,
    string? Vendor,
    DateOnly? ExpenseDate,
    string? Comment
);

File: shared/types/expense.ts
export interface Expense {
  id: string;
  monthId: string;
  categoryId: string;
  itemName: string;
  amount: number;
  vendor: string | null;
  expenseDate: string;
  comment: string | null;
  isRecurringInstance: boolean;
  isCompleted: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface ExpenseWithCategory extends Expense {
  categoryName: string;
  categoryIcon: string | null;
}

export interface CreateExpenseRequest {
  itemName: string;
  amount: number;
  categoryId: string;
  vendor?: string;
  expenseDate: string;
  comment?: string;
  isCompleted?: boolean;
}

export interface UpdateExpenseRequest {
  itemName?: string;
  amount?: number;
  categoryId?: string;
  vendor?: string;
  expenseDate?: string;
  comment?: string;
}
